sample-app/src/hooks/useSynchronizedRequest.tsx

```diff
@@ -1,18 +1,18 @@
 import { useRef, useState } from "react";
 
-export function useSynchronizedRequest<T>(
-  executeRequest: (inputValue?: string) => Promise<T | undefined>
+export function useSynchronizedRequest<RequestParams, ResponseType>(
+  executeRequest: (data?: RequestParams) => Promise<ResponseType | undefined>
 ): [
-    T | undefined,
-    (inputValue?: string) => Promise<T | undefined>
+    ResponseType | undefined,
+    (data?: RequestParams) => Promise<ResponseType | undefined>
```

Comment:
Do you think `RequestDataType` or `RequestParamType` might be more accurate? `RequestParams` is plural, but the `executeRequest` method has just one param

Reply:
I made it plural because there might be multiple parameters. But I understand what you mean that there is only one `data` param, so I'll change it to `RequestDataType`